### PR TITLE
Fix attribute params with regards to nullability

### PR DESCRIPTION
**Pull request by jcouv**
Fixes https://github.com/dotnet/roslyn/issues/36974

I'm also adding `[Obsolete]` on `NullableContextOptions` API which we intend to remove, but don't want to rush.

### CODE REVIEW

File: src/Compilers/Core/MSBuildTask/Csc.cs [hunk 1]
```diff
@@ -162,8 +162,10 @@ public string Nullable
             get { return (string)_store[nameof(Nullable)]; }
         }
 
+        [Obsolete("This property will be removed. Please use 'Nullable' instead")]
```
Comment (agocke) [inline src/Compilers/Core/MSBuildTask/Csc.cs]: What's going on here?

File: src/Compilers/Core/MSBuildTask/Csc.cs [hunk 2]
```diff
@@ -162,8 +162,10 @@ public string Nullable
...
         public string NullableContextOptions
```
Comment (agocke) [inline src/Compilers/Core/MSBuildTask/Csc.cs]: Let's remove it now. There's no better time to do it.

**Comment by jcouv** [comment]
FYI @rynowak We're removing the obsolete `Csc.NullableContextOptions` property (you guys are now using `Csc.Nullable`).

**Comment by rynowak** [comment]
cool beans!

**Comment by cston** [comment]
>             if (conversion.IsValid && conversion.Kind == ConversionKind.ImplicitReference)

`|| conversion.Kind == ConversionKind.Identity`

---
Refers to: src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs:925 in a174413. [](commit_id = a1744139131da472dd30c758195d87a48a86712e, deletion_comment = False)

**Pull request merged**